packages/cli/test/utils/simulation/LodestarBeaconNodeProcess.ts

```diff
@@ -0,0 +1,137 @@
+import {ChildProcess} from "node:child_process";
+import {mkdir, writeFile} from "node:fs/promises";
+import type {SecretKey} from "@chainsafe/bls/types";
+import {Api, getClient} from "@lodestar/api";
+import {nodeUtils} from "@lodestar/beacon-node/node";
+import {IChainForkConfig} from "@lodestar/config";
+import {IBeaconArgs} from "../../../src/cmds/beacon/options.js";
+import {getBeaconConfigFromArgs} from "../../../src/config/beaconParams.js";
+import {IGlobalArgs} from "../../../src/options/globalOptions.js";
+import {LodestarValidatorProcess} from "./LodestarValidatorProcess.js";
+import {BeaconNodeConstructor, BeaconNodeProcess, SimulationParams, ValidatorProcess} from "./types.js";
+import {BN_P2P_BASE_PORT, BN_P2P_REST_PORT, closeChildProcess, spawnProcessAndWait, __dirname} from "./utils.js";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const LodestarBeaconNodeProcess: BeaconNodeConstructor = class LodestarBeaconNodeProcess
+  implements BeaconNodeProcess {
+  static totalProcessCount = 0;
+  readonly params: SimulationParams;
+  readonly secretKeys: Record<number, SecretKey[]> = {};
+  readonly address: string;
+  readonly port: number;
+  readonly restPort: number;
+  readonly id: string;
+  api!: Api;
+  peerId!: string;
+  multiaddrs!: string[];
```

Comment:
Do not use the non-null assertion operator, find a better way to be typesafe. At least acknowledge that the type may be undefined and use `multiaddrs: string[] | undefined`